src/sources/closed_coil.cpp

```diff
@@ -0,0 +1,559 @@
+        *V_[i] = 0.0;
+        *J_[i] = 0.0;
+    }
+}
+
+void ClosedCoilSolver::SetBCs(){
+
+    high_terminal_.Append(elec_attrs_.first);
+    low_terminal_.Append(elec_attrs_.second);
+    high_src_ = new mfem::FunctionCoefficient(highV);
+    low_src_ = new mfem::FunctionCoefficient(lowV);
+
+    for (int i=0; i<2; ++i){
+
+        high_DBC_[i] = new hephaestus::FunctionDirichletBC(std::string("V_" + std::to_string(i)), high_terminal_, high_src_);
+        low_DBC_[i] = new hephaestus::FunctionDirichletBC(std::string("V_" + std::to_string(i)), low_terminal_, low_src_);
+        bc_maps_[i] = new hephaestus::BCMap;
+        bc_maps_[i]->Register("high_potential", high_DBC_[i], true);
+        bc_maps_[i]->Register("low_potential", low_DBC_[i], true);
+    }
+
+}
+
+void ClosedCoilSolver::SPSCurrent(){
+
+    for (int i=0; i<2; ++i){
+
+        fespaces_[i] = new hephaestus::FESpaces;
+        fespaces_[i]->Register(std::string("HCurl_" + std::to_string(i)), HCurlFESpace_[i], false);
+        fespaces_[i]->Register(std::string("H1_" + std::to_string(i)), H1FESpace_[i], false);
+
+        gridfunctions_[i] = new hephaestus::GridFunctions;
+        gridfunctions_[i]->Register(std::string("source_" + std::to_string(i)), J_[i], false);
+        gridfunctions_[i]->Register(std::string("V_" + std::to_string(i)), V_[i], false);
+
+        current_solver_options_[i] = new hephaestus::InputParameters;
+        current_solver_options_[i]->SetParam("Tolerance", float(1.0e-9));
+        current_solver_options_[i]->SetParam("MaxIter", (unsigned int)1000);
+        current_solver_options_[i]->SetParam("PrintLevel", 1);
+
+        sps_params_[i] = new hephaestus::InputParameters;
+        sps_params_[i]->SetParam("SourceName", std::string("source_" + std::to_string(i)));
+        sps_params_[i]->SetParam("PotentialName", std::string("V_" + std::to_string(i)));
+        sps_params_[i]->SetParam("HCurlFESpaceName", std::string("HCurl_" + std::to_string(i)));
+        sps_params_[i]->SetParam("H1FESpaceName", std::string("H1_" + std::to_string(i)));
+        sps_params_[i]->SetParam("SolverOptions", *current_solver_options_[i]);
+        sps_params_[i]->SetParam("ConductivityCoefName", std::string("magnetic_permeability"));
+
+        coefs_[i] = new hephaestus::Coefficients;
+        coefs_[i]->scalars.Register("magnetic_permeability", coef1_, false);
+
+        sps_[i] = new hephaestus::ScalarPotentialSource(*sps_params_[i]);
+        sps_[i]->Init(*gridfunctions_[i], *fespaces_[i], *bc_maps_[i], *coefs_[i]);
+
+        mfem::ParLinearForm dummy(HCurlFESpace_[i]);
+        sps_[i]->Apply(&dummy);
+
+        // Normalise the current through the wedges and use them as a reference
+        // The MPI allreduce is to take into account the MPI partitioning
+        double flux = calcJFlux(elec_attrs_.first, i);
+        double total_flux;
+        MPI_Allreduce(&flux, &total_flux, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+        *J_[i] /= abs(total_flux);
```

Comment:
Calculating the total flux across an internal surface is a useful operation for other postprocessors too - can we split this method out? So `calcFlux` takes in a boundary attribute (int) and a gridfunction (here `J[i]`) and returns the total flux.

I think this should also have an accompanying unit test for this method, to check the total flux is as expected for simple geometries/cross sections.

Reply:
I've changed the function to accept any GridFunction and a boundary attribute and calculates the flux of a vector field over it. Currently, I've put this new CalcFlux function outside of the ClosedCoilSolver class, although it is still declared and defined in the closed_coil .hpp and .cpp files. Perhaps there is a better home for this new function.

I'll start working on a test for this function!